pkg/packagekit/updates.jsx

```diff
@@ -800,12 +810,16 @@ class OsUpdates extends React.Component {
                             </div>
                         </div>);
                 } else {
+                    var num_updates = Object.keys(this.state.updates).length;
+                    var num_security_updates = count_security_updates(this.state.updates);
+
                     applyAll = (
                         <button className="btn btn-primary" onClick={ () => this.applyUpdates(false) }>
-                            {_("Install All Updates")}
+                            { num_updates == num_security_updates?
+                              _("Install Security Updates") : _("Install All Updates") }
                         </button>);
 
-                    if (this.state.haveSecurity) {
+                    if (num_updates > num_security_updates) {
```

Comment:
I think this still requires a check that `num_security_updates > 0`, to replace the previous `haveSecurity`. We don't want to show that button if there are only "normal" updates available.

Reply:
Indeed, ouch.  Fixed.